### PR TITLE
Remove anonymousTypeArguments. Enums use superclassConstructorParameters.

### PR DESCRIPTION
**Pull request by NightlyNexus**
~A test to demonstrate #332.~
~I'll update with the fix...~

Addresses #332 but also ended up refactoring and changing behavior. Happy to break into separate PRs.

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt [hunk 1]
```diff
@@ -95,7 +95,14 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
 
         codeWriter.emitCode("object")
         val supertype = if (superclass != ANY) {
-          listOf(CodeBlock.of(" %T(%L)", superclass, anonymousTypeArguments))
+          var superclassConstructorParameters = superclassConstructorParameters
+          if (anonymousTypeArguments.isNotEmpty()) {
+            superclassConstructorParameters = superclassConstructorParameters
+                .toMutableList().apply {
+              add(anonymousTypeArguments)
```
Comment (NightlyNexus) [inline src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: made this append, but I'd also like to get rid of `anonymousTypeArguments` completely, I think?

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt [hunk 2]
```diff
@@ -465,7 +462,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     }
 
     private fun ensureCanHaveSuperclass() {
-      check(kind.isSimpleClass || kind is Object) {
+      check(kind.isSimpleClass || kind.isEnum || kind is Object) {
         "only classes can have super classes, not $kind"
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: This message is starting to lose its accuracy.

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt [hunk 3]
```diff
@@ -422,17 +422,14 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     fun addAnnotation(annotation: KClass<*>) = addAnnotation(annotation.asClassName())
 
     fun addModifiers(vararg modifiers: KModifier) = apply {
-      check(anonymousTypeArguments == null) { "forbidden on anonymous types." }
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: Is there anything replacing these checks? I'm fine with just removing them, but it'd be nice to keep them

Comment (NightlyNexus) [reply src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: yes, i think we can do the same `isAnonymousClass` check from above. good idea!

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt (hunk 2 above)
Comment (NightlyNexus) [reply src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: whoops, saw that earlier and forgot to fix. brb.

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt (hunk 3 above)
Comment (JakeWharton) [reply src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: Would be nice to have tests for this in the future

Comment (NightlyNexus) [reply src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: #335

**Pull request merged**